File: web-app/src/lib/features/features/debug.ts
import type { Feature } from '../types';

export const debugFeatures = {
  'debug-create-snapshot': {
    id: 'debug-create-snapshot',
    title: 'Create Debug Snapshot',
    description: 'Creates a debug snapshot of the current session state for diagnostics.',
    rpcIds: ['debug:create-snapshot'],
    componentPaths: [],
    testIds: [
      'TestCreateDebugSnapshot_Succeeds',
      'TestCreateDebugSnapshot_WithNote',
    ],
    status: 'stable',
    since: '1.0.0',
  },
} as const satisfies Record<string, Feature>;

File: server/features/debug.go
package features

import "github.com/tstapler/stapler-squad/server/featureregistry"

var DebugCreateSnapshot = featureregistry.Feature{
	ID:          "debug-create-snapshot",
	Title:       "Create Debug Snapshot",
	Description: "Creates a debug snapshot of the current session state for diagnostics.",
	RPCIDs:      []string{"debug:create-snapshot"},
	Status:      featureregistry.StatusStable,
	Since:       "1.0.0",
}

func init() {
	featureregistry.Register(DebugCreateSnapshot)
}
